pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -455,6 +455,22 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 			return reconcile.Result{}, err
 		}
 
+		//Check if global pull secret is present
+		globalPullSecret, err := controllerutils.LoadSecretData(r.Client, cd.Spec.GlobalPullSecret.Name, cd.Namespace, corev1.DockerConfigJsonKey)
+		if err != nil {
+			cdLog.Warn("global pull secret not available")
+		}
+
+		// If global pull secret is present then merge the pullSecret and globalPullSecret and overwrite pullSecret
+		/* if globalPullSecret != nil {
+			// merge pullSecret and globalPullSecret
+			pullSecret, err = mergeJsons(globalPullSecret, pullSecret)
+			if err != nil {
+				cdLog.WithError(err).Error("unable to merge global pull secret with local pull secret")
+				return reconcile.Result{}, err
+			}
+		} */
```

Comment:
Looks correct would just be getting the secret name from an env var.

Reply:
@dgoodwin right, got to know about https://github.com/openshift/hive/blob/master/pkg/operator/hive/hive.go#L48-L50 from @csrwng

Reply:
@LalatenduMohanty the latest decision on the validation is that we simply won't make the field required anymore. So no additional changes to the admission webhook required.